src/haz3lcore/statics/Self.re

```diff
@@ -119,22 +119,27 @@ let of_exp_var = (ctx: Ctx.t, name: Var.t): exp =>
   | Some(var) => Common(Just(var.typ))
   };
 
-/* The self of a ctr depends on the ctx, but a
-   lookup failure doesn't necessarily means its
-   free; it may be given a type analytically */
-let of_ctr = (ctx: Ctx.t, name: Constructor.t, ty: Typ.t): t =>
-  switch (ty) {
-  | {term: Unknown(Internal), _} =>
-    IsConstructor({
-      name,
-      syn_ty:
-        switch (Ctx.lookup_ctr(ctx, name)) {
-        | None => None
-        | Some({typ, _}) => Some(typ)
-        },
-    })
-  | _ => IsConstructor({name, syn_ty: Some(ty)})
-  };
+let of_ctr =
```

Comment:
introducing a Mode dependency into Self feels weird, but I guess it makes the breakdown of the Self concept for anonymous sum types explicit. In any case this function has gotten pretty weird and awkward; probably deserves a comment